lib/core/commands/command_implementation/add_text_command.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:flutter/cupertino.dart';
+import 'package:paintroid/core/commands/command_implementation/graphic/graphic_command.dart';
+import 'package:paintroid/core/json_serialization/converter/paint_converter.dart';
+import 'package:equatable/equatable.dart';
+
+class AddTextCommand extends GraphicCommand with EquatableMixin {
+  final Offset point;
+  final String text;
+  final TextStyle style;
+
+  AddTextCommand(
+    this.point,
+    this.text,
+    this.style,
+    Paint paint,
+  ) : super(paint);
+
+  @override
+  void call(Canvas canvas) {
+    final textPainter = TextPainter(
+      text: TextSpan(text: text, style: style),
+      textDirection: TextDirection.ltr,
+    );
+
+    textPainter.layout(minWidth: 0, maxWidth: double.infinity);
+    final textOffset =
+        point - Offset(textPainter.width / 2, textPainter.height / 2);
+    textPainter.paint(canvas, textOffset);
+  }
+
+  void undo(Canvas canvas) {
+    // TODO
+  }
+
+  @override
+  List<Object?> get props => [point, text, paint];
+
+  @override
+  bool? get stringify => true;
+
+  @override
```

Comment:
This is not the way this project uses serialization. Please have a look at other commands. They are using a version strategy and we should use them in every command.